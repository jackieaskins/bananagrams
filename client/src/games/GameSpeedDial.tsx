import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  SpeedDial,
  SpeedDialAction,
  SpeedDialIcon,
} from "@mui/material";
import { ExitToApp } from "@mui/icons-material";
import { Link } from "react-router-dom";

import { useGameSpeedDial } from "./GameSpeedDialState";

const GameSpeedDial: React.FC = () => {
  const { leaveGameDialogOpen, showLeaveGameDialog, handleLeaveGameCancel } =
    useGameSpeedDial();

  return (
    <>
      <SpeedDial
        ariaLabel="Leave game speed dial"
        sx={{ position: "absolute", bottom: 10, right: 10 }}
        icon={<SpeedDialIcon />}
      >
        <SpeedDialAction
          icon={<ExitToApp />}
          tooltipTitle="Leave game"
          onClick={showLeaveGameDialog}
        />
      </SpeedDial>

      <Dialog open={leaveGameDialogOpen} onClose={handleLeaveGameCancel}>
        <DialogContent>
          <DialogContentText>
            Are you sure you want to leave the game?
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleLeaveGameCancel}>No</Button>
          <Button component={Link} to="/" color="primary">
            Yes
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

export default GameSpeedDial;
