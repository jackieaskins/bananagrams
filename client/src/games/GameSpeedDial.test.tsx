import { shallow } from "enzyme";
import GameSpeedDial from "./GameSpeedDial";

jest.mock("../styles", () => ({
  useStyles: () => ({
    drawerPaper: "drawerPaper",
  }),
}));

jest.mock("./GameSpeedDialState", () => ({
  useGameSpeedDial: () => ({
    leaveGameDialogOpen: true,
    showLeaveGameDialog: jest.fn().mockName("showLeaveGameDialog"),
    handleLeaveGameCancel: jest.fn().mockName("handleLeaveGameCancel"),
  }),
}));

describe("<GameSpeedDial />", () => {
  test("renders properly", () => {
    expect(shallow(<GameSpeedDial />)).toMatchSnapshot();
  });
});
