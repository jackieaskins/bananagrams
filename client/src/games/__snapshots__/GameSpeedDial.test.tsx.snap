// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameSpeedDial /> renders properly 1`] = `
<Fragment>
  <ForwardRef(SpeedDial)
    ariaLabel="Leave game speed dial"
    icon={<ForwardRef(SpeedDialIcon) />}
    sx={
      {
        "bottom": 10,
        "position": "absolute",
        "right": 10,
      }
    }
  >
    <ForwardRef(SpeedDialAction)
      icon={<Memo />}
      onClick={[MockFunction showLeaveGameDialog]}
      tooltipTitle="Leave game"
    />
  </ForwardRef(SpeedDial)>
  <ForwardRef(Dialog)
    onClose={[MockFunction handleLeaveGameCancel]}
    open={true}
  >
    <ForwardRef(DialogContent)>
      <ForwardRef(DialogContentText)>
        Are you sure you want to leave the game?
      </ForwardRef(DialogContentText)>
    </ForwardRef(DialogContent)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button)
        onClick={[MockFunction handleLeaveGameCancel]}
      >
        No
      </ForwardRef(Button)>
      <ForwardRef(Button)
        color="primary"
        component={
          {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        to="/"
      >
        Yes
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</Fragment>
`;
